### PR TITLE
Add contextual data selector filter

### PR DESCRIPTION
**Pull request by Kokan**
It makes possible to use `filter`s for selector.
The user could provide a filename `selector(filters(filter.con))`, where the `filters.conf` should be a configuration file, that has only the `version` and the `filter` context.

```
@version: 3.13
filter foo{ ... };
filter bar{ ... };
```

### CODE REVIEW

**Comment by Kokan** [comment]
@kira-syslogng retest this please test branch=fadd-contextual-data-selector-filter-ose;

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: fadd-contextual-data-selector-filter-ose and test suite: functions

**Comment by Kokan** [comment]
@kira-syslogng retest this please test branch=add-contextual-data-selector-filter-ose;

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: add-contextual-data-selector-filter-ose and test suite: functions

**Comment by lbudai** [comment]
I think the whole `parser_add_contextual_data_selector` section in the grammar should be removed.

**Comment by Kokan** [comment]
@kira-syslogng retest this please test branch=add-contextual-data-selector-filter-ose;

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: add-contextual-data-selector-filter-ose and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: add-contextual-data-selector-filter-ose and test suite: functions

File: modules/add-contextual-data/add-contextual-data-filter-selector.c [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+  if (!_populate_filter_store(self))
+    return FALSE;
+
+  self->filter_store = _filter_store_order_by_selectors(self->filter_store, ordered_selectors);
+
+  return TRUE;
+}
+
+static gchar *
+_resolve(AddContextualDataSelector *s, LogMessage *msg)
```
[the remark this answers is not in the document]
Comment (Kokan) [reply modules/add-contextual-data/add-contextual-data-filter-selector.c]: done

**Comment by lbudai** [comment]
`LL_CONTEXT_SELECTOR` should also be removed.

**Comment by lbudai** [comment]
Thanks!

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: add-contextual-data-selector-filter-ose and test suite: functions

**Pull request merged**